### DISCUSSION TITLE
Strokes without fill

### DISCUSSION THREAD

**Discussion by subchannel13** [Chlumsky/msdfgen#205] [Q&A] [answered]
Is there a set of command line options that do not fill closed shape and do not drop any stroke paths?

For instance consider this SVG file (a square with part of a diagonal inside):
![test](https://github.com/Chlumsky/msdfgen/assets/4396064/2196c13f-df96-494a-a534-df582a70ba14)
Is it possible to generate MSDF with a distance from those 5 lines (respecting sharp corners) as if no point is ever inside a shape?

**Comment by Chlumsky** [comment]
No, this is not supported by MSDFgen or the MSDF scheme itself since it treats all edges as infinite.

**Reply by subchannel13** [reply]
What do you mean by edges being infinite? In a direction perpendicular to the texture plain?

I've been playing with parameters and I got close enough to the ideal with `-fillrule negative` and by having overlapping "shapes" as separate images (both in input and output). I can live with that if there is no better way.
![output1](https://github.com/Chlumsky/msdfgen/assets/4396064/86cc6cc5-cab4-4547-805b-9ef542800fd4)
![output2](https://github.com/Chlumsky/msdfgen/assets/4396064/e506821f-7d22-4750-a2eb-37755102476f)

**Reply by Chlumsky** [reply, marked as answer]
The way MSDF works is that all edges continue indefinitely beyond their endpoints in the distance field and the corner is formed by the crossing of two edge segments in different channels. So if you have a line segment that suddenly ends, it would just continue in MSDF since there is no other edge to contain it. The entire scheme is designed only for contiguous regions with positive area, not one-dimensional lines.

**Reply by subchannel13** [reply]
Ah I get it/remember it from the paper now. But I will not stop abusing it for my use case :). Thanks for the awesome tool BTW!
